### PR TITLE
Add deprecated columns macro

### PR DESCRIPTION
**Pull request by boffbowsh**
Adds the deprecated columns macro from alphagov/whitehall. This allows
a grace period when making subtractive schema changes.

### CODE REVIEW

**Comment by benlovell** [comment]
Code changes seem good but the naming and description is a little out of line.

**Comment by boffbowsh** [comment]
Good point. Reworded the last commit message.

File: config/initializers/warn_on_unguarded_column_removals.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    rescue NameError
+      return "WARNING: Couldn't find model for table #{table_name}. I can't tell " +
+        "if column #{column_name} has been deprecated."
+    end
+
+    if klass.try(:deprecated_column_list).to_a.include?(column_name.to_s)
+      "OK: #{table_name}.#{column_name} has been deprecated."
+    else
+      "WARNING: #{table_name}.#{column_name} has not been deprecated. Please " +
+        "use the `DeprecatedColumns` mixin to avoid crashing Whitehall in " +
```
Comment (jamiecobbett) [inline config/initializers/warn_on_unguarded_column_removals.rb]: This shouldn't say "Whitehall".

Comment (boffbowsh) [inline config/initializers/warn_on_unguarded_column_removals.rb]: Good catch. Fixed.

Comment (benlovell) [inline config/initializers/warn_on_unguarded_column_removals.rb]: :blush:

**Pull request merged**